src/corefx/System.Globalization.Native/calendarData.cpp

```diff
@@ -6,11 +6,19 @@
 #include <string.h>
 #include <vector>
 
+#include "icushim.h"
 #include "config.h"
 #include "locale.hpp"
 #include "holders.h"
 #include "errors.h"
 
+#ifndef FEATURE_FIXED_ICU_VERSION
+// The System.Globalization.Native.so is built in mode that allows it to find the local
```

Comment:
It is half of the truth. The main problem is really the fact that it is not part of the stable APIs, so even if we wanted to support only newer versions of ICU that have that API, we should not do that unless it became part of the stable API.